crates/aptos-jwk-consensus/src/jwk_manager/mod.rs

```diff
@@ -103,15 +103,27 @@ impl JWKManager {
             .unwrap_or_default()
             .into_provider_vec()
             .into_iter()
-            .map(|provider| {
-                JWKObserver::spawn(
-                    self.epoch_state.epoch,
-                    self.my_addr,
-                    provider.name.clone(),
-                    provider.config_url.clone(),
-                    Duration::from_secs(10),
-                    local_observation_tx.clone(),
-                )
+            .flat_map(|provider| {
+                let OIDCProvider { name, config_url } = provider;
+                let maybe_issuer = String::from_utf8(name);
+                let maybe_config_url = String::from_utf8(config_url);
+                match (maybe_issuer, maybe_config_url) {
+                    (Ok(issuer), Ok(config_url)) => Some(JWKObserver::spawn(
+                        self.epoch_state.epoch,
+                        self.my_addr,
+                        issuer,
+                        config_url,
+                        Duration::from_secs(10),
+                        local_observation_tx.clone(),
+                    )),
+                    (maybe_issuer, maybe_config_url) => {
```

Comment:
I guess this syntax helps you avoid specifying all combinations of `None/Some`?

Reply:
exactly!